karton/gateway/operations.py

```diff
@@ -0,0 +1,464 @@
+    )
+    await send_success(websocket)
+
+
+async def generate_resource_download_urls(
+    task: Task, service_backend: KartonAsyncBackend
+) -> list[ResourceUrl]:
+    resources = {}
+
+    for resource in task.iterate_resources():
+        if resource.uid in resources:
+            continue
+        if resource.bucket != service_backend.default_bucket_name:
+            raise InvalidTaskError(
+                "Got task that references different bucket than the configured one "
+                "which is unsupported"
+            )
+        download_url = await service_backend.get_presigned_object_download_url(
+            bucket=service_backend.default_bucket_name, object_uid=resource.uid
+        )
+        resources[resource.uid] = ResourceUrl(uid=resource.uid, url=download_url)
+    return list(resources.values())
+
+
+@request_handler
+async def handle_get_task_request(
+    websocket: WebSocket, request: GetTaskRequest, session: UserSession
+) -> None:
+    if not session.is_bound:
+        raise InvalidBindError("Service has not declared consumer bind")
+
+    identity = session.identity
+    service_backend = await session.get_service_backend()
+
+    current_bind = service_backend.get_bind(identity)
+    if current_bind != session.karton_bind:
+        raise ExpiredBindError(
+            f"Consumer bind expired. "
+            f"Old binds: {str(session.karton_bind)}. "
+            f"New binds: {str(current_bind)}."
+        )
+
+    task = await service_backend.consume_routed_task(identity)
+    if not task:
+        raise OperationTimeoutError("No task found, try again")
+
+    try:
+        download_urls = await generate_resource_download_urls(task, service_backend)
+        task_token_info = TaskTokenInfo(
+            task_uid=task.uid,
+            resources=[r.uid for r in download_urls],
+        )
+        task_token = make_task_token(
+            task_token_info, gateway_config.secret_key, session.user.username
+        )
+        task_data = task.to_dict()
+        incoming_task = IncomingTask(
+            uid=task_data["uid"],
+            parent_uid=task_data["parent_uid"],
+            orig_uid=task_data["orig_uid"],
+            headers=task_data["headers"],
+            payload=task_data["payload"],
+            headers_persistent=task_data["headers_persistent"],
+            payload_persistent=task_data["payload_persistent"],
+            priority=task_data["priority"],
+        )
+        task_response_message = TaskResponseMessage(
+            task=incoming_task,
+            token=task_token,
+            download_urls=download_urls,
+        )
+        task_response = TaskResponse(message=task_response_message)
+        await websocket.send_json(task_response.model_dump(mode="json"))
+    except Exception:
+        # We need to crash gathered task if something went wrong in the process
+        exc_info = sys.exc_info()
+        task.error = traceback.format_exception(*exc_info)
```

Comment:
Is this different than
```suggestion
        task.error = traceback.format_exc()
```